### PR TITLE
jv-join

### PR DESCRIPTION
**Pull request by morozovyrv**

### CODE REVIEW

File: src/main/java/mate/jdbc/Main.java [hunk 1]
```diff
@@ -1,7 +1,74 @@
...
+
+        DriverService driverService = (DriverService) injector.getInstance(DriverService.class);
+        Driver driver1 = new Driver("Ivan Chernenko", "123456");
+        driver1 = driverService.create(driver1);
+        Driver driver2 = new Driver("Maksym Romanov", "654321");
+        driver2 = driverService.create(driver2);
+        Driver driver3 = new Driver("Roman Dzuba", "112233");
+        driver3 = driverService.create(driver3);
+        Driver driver4 = new Driver("Igor Shevchenko", "224311");
+        driver4 = driverService.create(driver4);
```
Comment (MaliukDaria) [inline src/main/java/mate/jdbc/Main.java:30]: It's bad practice to use numbers in variable names

File: src/main/java/mate/jdbc/dao/CarDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete car by id " + id + ".", e);
+        }
+    }
+
+    private List<Driver> getDriversForCar(Long id) {
+        String query = "SELECT * from drivers "
+                + "join cars_drivers "
+                + "on drivers.id = cars_drivers.driver_id "
+                + "where cars_drivers.car_id = ?";
```
Comment (MaliukDaria) [inline src/main/java/mate/jdbc/dao/CarDaoImpl.java:170]:
```suggestion
        String query = "SELECT * from drivers "
                + "JOIN cars_drivers "
                + "ON drivers.id = cars_drivers.driver_id "
                + "WHERE cars_drivers.car_id = ?";
```

File: src/main/java/mate/jdbc/dao/CarDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+                PreparedStatement statement =
+                        connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                drivers.add(getDriver(resultSet));
+            }
+            return drivers;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete car by id " + id + ".", e);
```
Comment (MaliukDaria) [inline src/main/java/mate/jdbc/dao/CarDaoImpl.java:182]: not informative exception message

File: src/main/java/mate/jdbc/dao/CarDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+                statement.setLong(2, driver.getId());
+                statement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Could not insert drivers to car" + car + ".", e);
+        }
+    }
+
+    private void removeDrivers(Car car) {
+        String query = "DELETE FROM `cars_drivers` WHERE car_id = ?";
```
Comment (MaliukDaria) [inline src/main/java/mate/jdbc/dao/CarDaoImpl.java:223]:
```suggestion
        String query = "DELETE FROM cars_drivers WHERE car_id = ?";
```

**Review by MaliukDaria — CHANGES_REQUESTED** [review]

File: src/main/java/mate/jdbc/dao/CarDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement =
+                        connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get list of cars.", e);
```
Comment (Nick97-git) [inline src/main/java/mate/jdbc/dao/CarDaoImpl.java:123]:
```suggestion
            throw new DataProcessingException("Couldn't get list of cars for driver: " + driver, e);
```

**Review by Nick97-git — CHANGES_REQUESTED** [review]

**Review by Nick97-git — APPROVED** [review]

**Pull request closed without merging**